### PR TITLE
Initial Death Rate and Temperature correlation

### PR DESCRIPTION
**Pull request by beca13**
Initial version.

### CODE REVIEW

File: HW_1.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import scipy
+import scipy.stats
+
+temp = pd.read_csv('avg_winter_temperature.csv')
+pneum = pd.read_csv('flu_pneumonia.CSV')
+
+merged = pd.merge(temp, pneum, on="state", how="inner")
+
+
+#Convert Celsius to Fahr --> Return Celsius conversion of input
```
Comment (Peja81) [inline HW_1.py]: move all functions definition after import * section.

File: HW_1.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+#new column "avg_fahreinheit"
+merged["avg_fahrenheit"] = celsius_to_fahr(merged["avg_celsius"])
+
+death_rate = merged.iloc[:, 2].values #independent variable vector
+death_abs = merged.iloc[:, 3].values #independent variable vector
+avg_fahrenheit = merged.iloc[:, 4].values   #dependent variable vector
+
+# create a function to interpret the results
+def interpret_corrcoef(value):
```
Comment (Peja81) [inline HW_1.py]: move above

File: HW_1.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+pneum = pd.read_csv('flu_pneumonia.CSV')
+
+merged = pd.merge(temp, pneum, on="state", how="inner")
+
+
+#Convert Celsius to Fahr --> Return Celsius conversion of input
+def celsius_to_fahr(temp_celsius):
+    temp_fahr = 9/5*temp_celsius + 32
+    return temp_fahr
+celsius_to_fahr(merged["avg_celsius"])
```
Comment (Peja81) [inline HW_1.py]: What does this line do?

File: HW_1.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    elif (value > 0.5 and value <=0.7):
+        print ('moderate positive')
+    elif (value > 0.7 and value <1):
+        print ('strong positive')
+    elif (value == 1):
+        print ('perfect positive')
+    else:
+        print ('error')
+        
+# correlation
```
Comment (Peja81) [inline HW_1.py]: Explain

File: HW_1.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+personRResult=scipy.stats.pearsonr(avg_fahrenheit, death_abs)      
+interpret_corrcoef(personRResult[0])
+sns.regplot( data=merged, x='avg_fahrenheit', y='death_abs')
+
+# correlation
+personRResult1=scipy.stats.pearsonr(avg_fahrenheit, death_rate)
```
Comment (Peja81) [inline HW_1.py]: avoid adding suffixes 1, 2  etc..
Use proper names e.g. personScoreFarhDeathRate

File: HW_1.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    elif (value > 0.7 and value <1):
+        print ('strong positive')
+    elif (value == 1):
+        print ('perfect positive')
+    else:
+        print ('error')
+        
+# correlation
+personRResult=scipy.stats.pearsonr(avg_fahrenheit, death_abs)      
+interpret_corrcoef(personRResult[0])
```
Comment (Peja81) [inline HW_1.py]: explain why [0]

File: HW_1.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Tue May 21 17:46:24 2019
+
+@author: brank_000
+"""
+#Correlation between
```
Comment (Peja81) [inline HW_1.py]: Specify type of correlation and which test is going to be used here.

File: HW_1.py [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+personRResult=scipy.stats.pearsonr(avg_fahrenheit, death_abs)      
+interpret_corrcoef(personRResult[0])
+sns.regplot( data=merged, x='avg_fahrenheit', y='death_abs')
+
+# correlation
+personRResult1=scipy.stats.pearsonr(avg_fahrenheit, death_rate)      
+interpret_corrcoef(personRResult1[0])
+plt.figure(); # create new window / blank chart
+sns.regplot( data=merged, x='avg_fahrenheit', y='death_rate')
+
```
Comment (Peja81) [inline HW_1.py]: add trivial example of correlation between Fahrenheit and Celsius values.

**Review by Peja81 — CHANGES_REQUESTED** [review]

**Comment by Peja81** [comment]
Upload also CSV files. and update readme.md

**Comment by beca13** [comment]
Rename HW_1 to something more meaningful

File: HW_1.py (hunk 8 above)
Comment (beca13) [reply HW_1.py]: 24 line

File: HW_1.py (hunk 6 above)
Comment (beca13) [reply HW_1.py]: ??

File: DeathRateAndTemperatureCorr.py [hunk 9]
```diff
@@ -1,69 +1,74 @@
...
+import matplotlib.pyplot as plt
+import seaborn as sns
+import pandas as pd
+import scipy
+import scipy.stats
+
+temp = pd.read_csv('avg_winter_temperature.csv')
+pneum = pd.read_csv('flu_pneumonia.CSV')
+
+merged = pd.merge(temp, pneum, on="state", how="inner")
```
Comment (Peja81) [inline DeathRateAndTemperatureCorr.py]: Add comment explaining csv structure for both files and key field (used for join)

File: DeathRateAndTemperatureCorr.py [hunk 10]
```diff
@@ -1,69 +1,74 @@
...
+
+
+#convert Celsius to Fahrenhait --> Return Celsius conversion of input
+#math correlation between F and C --> F = C × 1.8 + 32 
+def celsius_to_fahr(temp_celsius):
```
Comment (Peja81) [inline DeathRateAndTemperatureCorr.py:15]: move this function under import (easier to read execution process)

File: DeathRateAndTemperatureCorr.py [hunk 11]
```diff
@@ -1,69 +1,74 @@
...
+#new column "avg_fahreinheit"
+merged["avg_fahrenheit"] = celsius_to_fahr(merged["avg_celsius"])
+
+
+#create a function to interpret the results
+death_rate = merged.iloc[:, 2].values       #independent variable vector
+death_abs = merged.iloc[:, 3].values        #independent variable vector
+avg_fahrenheit = merged.iloc[:, 4].values   #dependent variable vector
+
+def interpret_corrcoef(value):
```
Comment (Peja81) [inline DeathRateAndTemperatureCorr.py:19]: move functions up - right below import

File: DeathRateAndTemperatureCorr.py [hunk 12]
```diff
@@ -1,69 +1,74 @@
...
+    elif (value > 0.7 and value <1):
+        print ('strong positive')
+    elif (value == 1):
+        print ('perfect positive')
+    else:
+        print ('error')# if value is not between -1 and 1
+        
+#plot --> correlation between the number of deaths-absolute associated with influenza and the average temperature
+personScoreFahrDeathAbs=scipy.stats.pearsonr(avg_fahrenheit, death_abs)      
+interpret_corrcoef(personScoreFahrDeathAbs[0])
```
Comment (Peja81) [inline DeathRateAndTemperatureCorr.py:68]: print to console which correlation is being interpreted here

File: DeathRateAndTemperatureCorr.py [hunk 13]
```diff
@@ -1,69 +1,74 @@
...
+sns.regplot( data=merged, x='avg_fahrenheit', y='death_abs')
+
+#plot --> correlation between the number of deaths-rate associated with influenza and the average temperature
+personScoreFahrDeathRate=scipy.stats.pearsonr(avg_fahrenheit, death_rate)      
+interpret_corrcoef(personScoreFahrDeathRate[0])
```
Comment (Peja81) [inline DeathRateAndTemperatureCorr.py:76]: print to console which correlation is being interpreted here

File: DeathRateAndTemperatureCorr.py [hunk 14]
```diff
@@ -1,69 +1,74 @@
...
-interpret_corrcoef(personRResult1[0])
-plt.figure(); # create new window / blank chart
-sns.regplot( data=merged, x='avg_fahrenheit', y='death_rate')
-
+# -*- coding: utf-8 -*-
+"""
+Created on Tue May 21 17:46:24 2019
+
+@author: brank_000
+"""
```
Comment (Peja81) [inline DeathRateAndTemperatureCorr.py]: remove this

File: DeathRateAndTemperatureCorr.py (hunk 9 above)
Comment (Peja81) [reply DeathRateAndTemperatureCorr.py]: Explain every column and especially why do we need to convert to fahreneit

File: DeathRateAndTemperatureCorr.py [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+#plot --> correlation between the number of deaths-absolute associated with influenza and the average temperature
+personScoreFahrDeathAbs=scipy.stats.pearsonr(avg_fahrenheit, death_abs)      
+interpret_corrcoef(personScoreFahrDeathAbs[0])
+sns.regplot( data=merged, x='avg_fahrenheit', y='death_abs')
+
+#plot --> correlation between the number of deaths-rate associated with influenza and the average temperature
+personScoreFahrDeathRate=scipy.stats.pearsonr(avg_fahrenheit, death_rate)      
+interpret_corrcoef(personScoreFahrDeathRate[0])
+plt.figure() # create new window / blank chart
+sns.regplot( data=merged, x='avg_fahrenheit', y='death_rate')
```
Comment (Peja81) [inline DeathRateAndTemperatureCorr.py]: set proper title of the graph (for both evaluations)

**Review by Peja81 — CHANGES_REQUESTED** [review]

File: HW_1.py (hunk 6 above)
Comment (Peja81) [reply HW_1.py]: put what is in the first column and what is in the second element of the result. Check scipy.stats.pearsonr function (online)

**Review by Peja81 — APPROVED** [review]

**Pull request merged**